retrace-server.spec.in

```diff
@@ -101,6 +101,36 @@ rm -f %{buildroot}%{_infodir}/dir
 %define retrace_gid_uid 174
 getent group retrace > /dev/null || groupadd -f -g %{retrace_gid_uid} --system retrace
 getent passwd retrace > /dev/null || useradd --system -g retrace -u %{retrace_gid_uid} -b %{_sharedstatedir} -m -s /sbin/nologin retrace
+if grep -qs '^[a-z_][a-z0-9_]\{0,30\}:[0-9]\+:[0-9]\+\s*$' /etc/subuid; then
+    if ! grep -q '^retrace:' /etc/subuid; then
+        SUBUID=$(grep '^[a-z_][a-z0-9_]\{0,30\}:[0-9]\+:[0-9]\+\s*$' /etc/subuid \
+                 | sed 's/^[^:]*://' \
+                 | sort -n \
+                 | tail -n 1)
+        SUBUID_CNT=$(echo "$SUBUID" | sed -e 's/.*://' -e 's/\s*$//')
+        SUBUID=$(echo "$SUBUID" | sed 's/:.*//')
+        RETRACE_SUBUID_ENTRY=retrace:$((SUBUID+SUBUID_CNT+1)):65536
+        echo $RETRACE_SUBUID_ENTRY >> /etc/subuid
+    fi
+else
+    RETRACE_SUBUID_ENTRY=retrace:100000:65536
+    echo $RETRACE_SUBUID_ENTRY >> /etc/subuid
+fi
+if grep -qs '^[a-z_][a-z0-9_]\{0,30\}:[0-9]\+:[0-9]\+\s*$' /etc/subgid; then
```

Comment:
I’m kind of feeling dubious about this now. I don’t think the chance of the data being malformed is very high, so I would probably just stick with and `cut`.

Reply:
Yeah, it did feel like a bit of an overkill when I was writing it. Rewritten.